util/envDynawo.sh

```diff
@@ -769,6 +770,50 @@ build_3rd_party_version() {
   fi
 }
 
+build_3rd_party_version_cmake() {
+  CMAKE_OPTIONNAL=""
+  if [ $DYNAWO_BOOST_HOME_DEFAULT != true ]; then
+    CMAKE_OPTIONNAL="-DBOOST_ROOT=$DYNAWO_BOOST_HOME"
+  fi
+  if [ $DYNAWO_ZLIB_HOME_DEFAULT != true ]; then
+    CMAKE_OPTIONNAL="$CMAKE_OPTIONNAL -DZLIB_ROOT=$DYNAWO_ZLIB_HOME"
+  fi
+  if [ $DYNAWO_LIBARCHIVE_HOME_DEFAULT != true ]; then
+    CMAKE_OPTIONNAL="$CMAKE_OPTIONNAL -DLIBARCHIVE_HOME=$DYNAWO_LIBARCHIVE_HOME"
+  fi
+  case $DYNAWO_BUILD_TYPE in
+    Tests|TestCoverage)
+      if [ $DYNAWO_GTEST_HOME_DEFAULT != true ]; then
+        CMAKE_OPTIONNAL="$CMAKE_OPTIONNAL -DGTEST_ROOT=$DYNAWO_GTEST_HOME"
+      fi
+      if [ $DYNAWO_GMOCK_HOME_DEFAULT != true ]; then
+        CMAKE_OPTIONNAL="$CMAKE_OPTIONNAL -DGMOCK_HOME=$DYNAWO_GMOCK_HOME"
+      fi
+      ;;
+    *)
+      ;;
+  esac
+
+  mkdir -p $DYNAWO_THIRD_PARTY_BUILD_DIR_VERSION/build
+  cd $DYNAWO_THIRD_PARTY_BUILD_DIR_VERSION/build
+  cmake $DYNAWO_SRC_DIR/3rdParty \
+    -DCMAKE_INSTALL_PREFIX=$DYNAWO_THIRD_PARTY_INSTALL_DIR_VERSION \
+    -DDOWNLOAD_DIR=$DYNAWO_THIRD_PARTY_BUILD_DIR_VERSION/src \
+    -DTMP_DIR=$DYNAWO_THIRD_PARTY_BUILD_DIR_VERSION/tmp \
+    -DCMAKE_C_COMPILER=$DYNAWO_C_COMPILER \
+    -DCMAKE_CXX_COMPILER=$DYNAWO_CXX_COMPILER \
+    -DCXX11_ENABLED=$DYNAWO_CXX11_ENABLED \
+    -DCMAKE_BUILD_TYPE=$DYNAWO_BUILD_TYPE \
+    -DOPENMODELICA_INSTALL=$DYNAWO_INSTALL_OPENMODELICA \
+    -DOPENMODELICA_SRC=$DYNAWO_SRC_OPENMODELICA \
+    -DBUILD_SHARED_LIBS=$(if [ "$DYNAWO_LIBRARY_TYPE" = "SHARED" ]; then echo -n "ON"; else echo -n "OFF"; fi) \
+    -G "$DYNAWO_CMAKE_GENERATOR" \
+    $CMAKE_OPTIONNAL || return 1
+  make -j $DYNAWO_NB_PROCESSORS_USED
+  RETURN_CODE=$?
```

Comment:
Shouldn't we remove the tmp dir if the compilation passed?

Reply:
which tmp dir ?

Reply:
It is not really a tmp dir, it is a cmake folder to build external project and it is named as tmp in its internal working, no need to delete it